### DISCUSSION TITLE
Can I set `^N` to toggle type-to-nav for all context?

### DISCUSSION THREAD

**Discussion by amalgame21** [jarun/nnn#1660] [General]
Now it just toggle type-to-nav mode for current context only, the mode of previously opened context does not affected.
And the type-to-nav mode of the to-be-opened context depends on the mode on the current context.

Can I somehow to make it behave like a global switch? 
If it involves some minor code modification, where should I pay attention to? Thank!

**Comment by jarun** [comment]
In the following lines, you would have to toggle the filtermode for all contexts.

    7330                         case SEL_MFLTR:
    7331                                 cfg.filtermode ^= 1;

It is a per-context property.

**Reply by amalgame21** [reply]
Thank you for your valuable tip !
After hours looking at the source code I finally come up with a solution for that by manipulating `cfg.filtermode` (as you suggested), which not only works for `^N`, but also the mouse click toggle (didn't know about this function before searching into the source code). 
Thank you very much !

```diff
diff --git a/src/nnn.c b/src/nnn.c
--- a/src/nnn.c
+++ b/src/nnn.c
@@ -4269,6 +4269,7 @@ static void savecurctx(char *path, char *curname, int nextctx)
 {
 	settings tmpcfg = cfg;
 	context *ctxr = &g_ctx[nextctx];
+	uint_t r = cfg.filtermode;
 
 	/* Save current context */
 	if (curname)
@@ -4297,6 +4298,7 @@ static void savecurctx(char *path, char *curname, int nextctx)
 
 	tmpcfg.curctx = nextctx;
 	cfg = tmpcfg;
+	cfg.filtermode = r;
 }
 
 #ifndef NOSSN

```
